src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -363,6 +376,62 @@ public Pose2d getVirtualTarget() {
             .inverse());
   }
 
+  /**
+   * Gets the pose at some time in the future
+   *
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getFuturePose(double time) {
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                getRobotRelativeSpeeds().vxMetersPerSecond * time,
+                getRobotRelativeSpeeds().vyMetersPerSecond * time,
+                new Rotation2d()));
+  }
+
+  @AutoLogOutput(key = "AutoAim/FuturePose")
+  public Pose2d getFuturePose() {
+    return getFuturePose(AutoAim.LOOKAHEAD_TIME);
+  }
+
+  /**
+   * Faces the robot towards a translation on the field
+   *
+   * @param xMetersPerSecond Requested X velocity
+   * @param yMetersPerSecond Requested Y velocity
+   * @param time Time in the future to point from
+   * @return A command refrence that rotates the robot to a computed rotation
+   */
+  public Command pointTowardsTranslation(
+      DoubleSupplier xMetersPerSecond, DoubleSupplier yMetersPerSecond, double time) {
+    ProfiledPIDController headingController =
+        // assume we can accelerate to max in 0.5 seconds
+        new ProfiledPIDController(
+            1.0, 0.0, 0.0, new Constraints(MAX_ANGULAR_SPEED, MAX_ANGULAR_SPEED / 0.5));
+    headingController.enableContinuousInput(-Math.PI, Math.PI);
+
+    return this.runVelocityFieldRelative(
+            () -> {
+              double feedbackOutput =
+                  headingController.calculate(
+                      getFuturePose(time).getRotation().getRadians(),
+                      getRotationToTranslation(getVirtualTarget()).getRadians());
+
+              return new ChassisSpeeds(
+                  xMetersPerSecond.getAsDouble(),
+                  yMetersPerSecond.getAsDouble(),
+                  feedbackOutput + headingController.getSetpoint().velocity);
+            })
+        .beforeStarting(
+            () ->
+                headingController.reset(
```

Comment:
make sure youre passing in the right rotation to reset, i believe it should be your current rotation and velocity